apache/superset
# feat(plugin-chart-echarts): support non-timeseries x-axis

### SUMMARY
A first step in the direction of adding support for non-temporal x-axes to ECharts line/area/bar/step/scatter charts. When the feature flag is enabled, a new control "X-Axis" shows up which accepts any type of column, be it temporal, numerical or categorical. For new charts the control is mandatory, but old (temporal) charts will still work. Other changes:
- Remove the "Time-series" prefix from all ECharts timeseries charts when 
- make formula annotations work with numeric x-axes
- clean up some TS types/interfaces that had fallen out of sync with the post processing API in the backend

### SCREENSHOTS
When the feature flag is enabled, the chart picker now shows all old ECharts time-series charts without the prefix (the descriptions are also updated to reflect that the chart is now "generic"):
![image](https://user-images.githubusercontent.com/33317356/148917030-02acb68d-5169-4ea4-82df-baf39f86e73a.png)

Bar chart v2 (previously Time-series bar chart) now works with non-temporal x-axis:
![image](https://user-images.githubusercontent.com/33317356/148917393-17d190e8-dcd9-439d-aafc-94795fce034a.png)

Another example; Area chart v2, with the x-axis labels rotated 45 degrees (previously Time-series area chart):
![image](https://user-images.githubusercontent.com/33317356/148918335-1f26c0e9-3f7e-4262-8526-0eca70630852.png)

### VIDEO
https://user-images.githubusercontent.com/33317356/148947353-9a027210-a9e4-42ab-93b5-bf356261f359.mp4

### TESTING INSTRUCTIONS
1. Enable the `GENERIC_CHART_AXES` feature flag
2. Open the chart picker and notice all "Time-series" charts have been replaced with a more generic name "Time-series line chart" - > "Line chart"
3. Drop a categorical column in the "x axis" control
4. Notice that a line chart is created with a non-temporal x-axis

### ADDITIONAL INFORMATION


- [x] Has associated issue: closes #6568
